homeassistant/components/sensor/recorder.py

```diff
@@ -226,9 +229,36 @@ def _normalize_states(
     return DEVICE_CLASS_UNITS[key], fstates
 
 
-def reset_detected(state: float, previous_state: float | None) -> bool:
+def warn_dip(hass: HomeAssistant, entity_id: str) -> None:
+    """Log a warning once if a sensor with state_class_total has a decreasing value."""
+    if WARN_DIP not in hass.data:
+        hass.data[WARN_DIP] = set()
+    if entity_id not in hass.data[WARN_DIP]:
+        hass.data[WARN_DIP].add(entity_id)
+        domain = entity_sources(hass).get(entity_id, {}).get("domain")
+        if domain in ["energy", "growatt_server", "solaredge"]:
+            return
+        _LOGGER.warning(
```

Comment:
Do we want to differentiate entities from custom integrations and/or the template/rest integrations here?

Reply:
We want to understand how common this is regardless of if it's a custom integration or not.
Not sure about template and mqtt though, we could filter them out or try to understand the use case?

Reply:
Let's do all now, and adjust later if needed :+1: